### PR TITLE
Add GitHub action for Ansible sanity tests

### PR DESCRIPTION
**Pull request by verkaufer**
# Summary

Adds automated `ansible-test sanity` and `ansible-test units` GitHub workflows. The `ansible-test sanity` job is run for each version of `ansible-core` supported by the collection. 

"Sanity" tests are equivalent to lint tests, and do not require a Connect server to run.

**Supported `ansible-core` versions**:
- 2.9 (⚠️ deprecated beginning **December 2022**)
- 2.10
- 2.11
- 2.12 

The GitHub action will run a sanity test suite for each ansible version listed above.

### References

This workflow was heavily inspired by [T-Systems-MMS/ansible-collection-icinga-directory](https://github.com/T-Systems-MMS/ansible-collection-icinga-director/blob/35ce15a5529d9a3f23dcfb671c02147747475813/.github/workflows/main.yml).

### CODE REVIEW

**Comment by verkaufer** [comment]
TODO:
- add caching for the python setup and checkout steps?

File: .github/workflows/ansible-tests.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  # arbitrarily choosing the latest stable version to run ansible-test units
+  # running 'ansible-test units' for each ansible+python version isn't necessary
+  ANSIBLE_FOR_UNIT_TESTS: "2.11"
+  # defines where ansible will look for the collection
+  # must be a sub-directory of ansible_collections/
+  COLLECTION_PATH: ansible_collections/onepassword/connect
+
+jobs:
+  ansible-sanity:
+    name: Ansible Linter (v${{ matrix.ansible }}, py${{ matrix.python }})
```
Comment (jpcoenen) [inline .github/workflows/ansible-tests.yml:20]: That it is called `Ansible Linter` here threw me a bit off when looking at the `Checks` tab. I was looking for `Ansible Sanity Test`

image

```suggestion
    name: Ansible Sanity Test (v${{ matrix.ansible }}, py${{ matrix.python }})
```

**Review by jpcoenen — APPROVED** [review]
Awesome work! 🙌 

One minor remark. Apart from that, looks like a great improvement to the workflow.

File: .github/workflows/ansible-tests.yml (hunk 1 above)
Comment (verkaufer) [reply .github/workflows/ansible-tests.yml:20]: Good call out! I'll make that change

**Pull request merged**